packages/node/src/methods/install-operations.ts

```diff
@@ -82,8 +104,175 @@ export async function addAppInstance(
+    appInstance,
+    appInstanceInfo.myDeposit,
+    appInstanceInfo.peerDeposit
+  );
+
+  await store.updateChannelWithAppInstanceInstallation(
+    updatedStateChannel,
+    appInstance,
+    appInstanceInfo
+  );
+
+  return appInstanceInfo;
+}
+
+/**
+ * @param appInstanceInfo The AppInstanceInfo to convert
+ * @param channel The channel the AppInstanceInfo belongs to
+ */
+function createAppInstanceFromAppInstanceInfo(
+  proposedAppInstanceInfo: ProposedAppInstanceInfo,
+  channel: StateChannel
+): AppInstance {
+  const appFunctionSigHashes = getAppFunctionSigHashes(proposedAppInstanceInfo);
+
+  const appInterface: AppInterface = {
+    addr: proposedAppInstanceInfo.appId,
+    applyAction: appFunctionSigHashes.applyAction,
+    resolve: appFunctionSigHashes.resolve,
+    getTurnTaker: appFunctionSigHashes.getTurnTaker,
+    isStateTerminal: appFunctionSigHashes.isStateTerminal,
+    stateEncoding: proposedAppInstanceInfo.abiEncodings.stateEncoding,
+    actionEncoding: proposedAppInstanceInfo.abiEncodings.actionEncoding
+  };
+
+  // TODO: throw if asset type is ETH and token is also set
+  const terms: Terms = {
+    assetType: proposedAppInstanceInfo.asset.assetType,
+    limit: proposedAppInstanceInfo.myDeposit.add(
+      proposedAppInstanceInfo.peerDeposit
+    ),
+    token: proposedAppInstanceInfo.asset.token
+      ? proposedAppInstanceInfo.asset.token
+      : AddressZero
+  };
+
+  return new AppInstance(
+    channel.multisigAddress,
+    // TODO: generate ephemeral app-specific keys
+    channel.multisigOwners,
+    proposedAppInstanceInfo.timeout.toNumber(),
+    appInterface,
+    terms,
+    // TODO: pass correct value when virtual app support gets added
+    false,
+    // TODO: this should be thread-safe
+    channel.numInstalledApps,
+    channel.rootNonceValue,
+    proposedAppInstanceInfo.initialState,
+    0,
+    proposedAppInstanceInfo.timeout.toNumber()
+  );
+}
+
+function getAppFunctionSigHashes(
+  appInstanceInfo: AppInstanceInfo
+): AppInterfaceSighashParameters {
+  const stateEncoding = appInstanceInfo.abiEncodings.stateEncoding;
+
+  const resolveSigHash = new Interface([`resolve(${stateEncoding}, ${TERMS})`])
+    .functions.resolve.sighash;
+
+  const getTurnTakerSigHash = new Interface([`getTurnTaker(${stateEncoding})`])
+    .functions.getTurnTaker.sighash;
+
+  const isStateTerminalSigHash = new Interface([
+    `isStateTerminal(${stateEncoding})`
+  ]).functions.isStateTerminal.sighash;
+
+  let applyActionSigHash = "0x00000000";
+  if (appInstanceInfo.abiEncodings.actionEncoding !== undefined) {
+    applyActionSigHash = new Interface([
+      `applyAction(${stateEncoding}, ${
+        appInstanceInfo.abiEncodings.actionEncoding
+      })`
+    ]).functions.applyAction.sighash;
   }
+
+  return {
+    resolve: resolveSigHash,
+    applyAction: applyActionSigHash,
+    getTurnTaker: getTurnTakerSigHash,
+    isStateTerminal: isStateTerminalSigHash
+  } as AppInterfaceSighashParameters;
```

Comment:
```suggestion
  };
```

You can set the return type as `AppInterfaceSighashParameters` and it will type check this object automatically. In this scenario doing type casting actually _weakens_ the type system because `{ blah: blah} as AppInterfaceSighashParameters` would not throw a type error.